lib/puppet/provider/mysql.rb

```diff
@@ -38,11 +38,35 @@ class Puppet::Provider::Mysql < Puppet::Provider
   ].join(':')
 
   # rubocop:disable Style/HashSyntax
-  commands :mysql_raw  => 'mysql'
-  commands :mysqld     => 'mysqld'
-  commands :mysqladmin => 'mysqladmin'
+  commands :mysql_client     => 'mysql'
+  commands :mariadb_client   => 'mariadb'
+  commands :mysqld_service   => 'mysqld'
+  commands :mariadbd_service => 'mariadbd'
+  commands :mysql_admin      => 'mysqladmin'
+  commands :mariadb_admin    => 'mariadb-admin'
   # rubocop:enable Style/HashSyntax
 
+  def self.mysql_raw(*args)
+    if newer_than('mariadb' => '11.0.0') && mysqld_version_string.scan(%r{mariadb}i)
+      return mariadb_client(*args)
+    end
+    mysql_client(*args)
```

Comment:
Also, I think we can assume Ruby 2.7 and use argument forwarding.

Once you've defined them as optional, I wonder if this works:

```suggestion
  def self.mysql_raw(...)
    command(:mariadb_client) ? mariadb_client(...) : mysql_client(...)
```